proposals/0000-a11y-cross-platform.md

```diff
@@ -0,0 +1,133 @@
+---
+title: Cross-platform Accessibility Improvements
+author:
+- Marc Mulcahy
+date: 2018-10-31
+---
+
+# RFC0000: Cross-platform Accessibility Improvements
+
+## Summary
+
+Several accessibility-related properties are only available on one platform (either iOS or Android, but not both). This proposal attempts to make as many properties available on both Android and iOS as possible.
+
+Properties only available on iOS include:
+
+* accessibilityViewIsModal
+* accessibilityElementsHidden
+
+Properties only available on Android include:
+
+* accessibilityLiveRegion
+
+Several properties are missing from both platforms:
+
+* accessibilityCompoundComponent: replace accessible-- indicates to assistive technologies that all the components inside this one should be treated as a single component.
+* accessibilityRelationships: describes how this component is related to others
+* accessibilityRange: expose information about range-based components such as sliders
```

Comment:
I think we can probably learn from web here and expose the range props to the underlying accessibility layer of the platform. Or model this off ARIA props for progressbar etc

Reply:
We've found that in practice, exposing range information in a consumable way is fairly difficult. Exposing a minimum, current, and maximum value allow you to speak a percentage, which is fine for some types of progress bars. But for others, where the actual value is relevant (say a color contrast setting, stereo balance control, movie timeline slider, etc.) it would be nice to include some information that could cause the screen reader to speak either a string or the current value instead of a percentage. Slider's also imply programatic actions to adjust them.

Reply:
> But for others, where the actual value is relevant (say a color contrast setting, stereo balance control, movie timeline slider, etc.) it would be nice to include some information that could cause the screen reader to speak either a string or the current value instead of a percentage.

ARIA has a property for that `aria-valuetext` - https://www.w3.org/TR/wai-aria-practices/examples/slider/slider-2.html

I do like how understandable and extensible the `accessibilityRelationships` and `accessibilityRange` props names are vs having many more individual props ARIA.

Are you thinking it would look something like this (property names tbh for relationships)?

```
<View
  accessibilityRange={{ min, max, current, text }}
  accessibilityRelationships={{ labelID, descriptionID, focusID }}
/>
```